scripts/check_phase42_boundary_correspondence.py

```
@@ -0,0 +1,1074 @@
+)
+
+
+class Phase42Error(Exception):
+    """Raised when a supplied artifact fails the Phase42 source checks."""
+
+
+def load_json(path: pathlib.Path) -> dict[str, Any]:
+    try:
+        data = json.loads(path.read_text(encoding="utf-8"))
+    except json.JSONDecodeError as exc:
+        raise Phase42Error(f"{path}: invalid JSON: {exc}") from exc
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)